bundles/org.openhab.core.automation.module.script/src/main/resources/OH-INF/config/script-profile.xml

```diff
@@ -11,12 +11,25 @@
 				may return null to discard the updates/commands and not pass them through.</description>
 			<limitToOptions>false</limitToOptions>
 		</parameter>
-		<parameter name="toHandlerScript" type="text">
-			<label>Item To Thing Transformation</label>
+		<parameter name="commandFromItemScript" type="text">
+			<label>Item Command To Thing Transformation</label>
 			<description>The Script for transforming commands from the item to the Thing handler. The script may return null to
 				discard the commands and not pass them through.</description>
 			<limitToOptions>false</limitToOptions>
 		</parameter>
+		<parameter name="stateFromItemScript" type="text">
+			<label>Item State To Thing Transformation</label>
+			<description>The Script for transforming states from the item to the Thing handler. The script may return null to
+				discard the states and not pass them through.</description>
+			<limitToOptions>false</limitToOptions>
+		</parameter>
+		<parameter name="toHandlerScript" type="text">
```

Comment:
Maybe show this one as "advanced"? That hides it for new users, while keeping it for this that already have it configured.

Side note: It would be great if Main UI would automatically show "advanced" parameters if something is configured there. This would also be helpful e.g. for HTTP binding things.

Reply:
> Side note: It would be great if Main UI would automatically show "advanced" parameters if something is configured there. This would also be helpful e.g. for HTTP binding things.

There we go: https://github.com/openhab/openhab-webui/pull/2313